src/goto-harness/goto_harness_parse_options.cpp

```diff
@@ -0,0 +1,60 @@
+/******************************************************************\
+
+Module: goto_harness_parse_options
+
+Author: Diffblue Ltd.
+
+\******************************************************************/
+
+#include <cstddef>
+#include <iostream>
+#include <string>
+
+#include <util/exit_codes.h>
+#include <util/version.h>
+
+#include "goto_harness_parse_options.h"
+
+int goto_harness_parse_optionst::doit()
+{
+  if(cmdline.isset("version"))
+  {
+    std::cout << CBMC_VERSION << '\n';
+    return CPROVER_EXIT_SUCCESS;
+  }
+
+  help();
+  return CPROVER_EXIT_USAGE_ERROR;
+}
+
+void goto_harness_parse_optionst::help()
+{
+  auto align_center_with_border = [](const std::string &text) {
+    auto const total_length = std::size_t{63};
+    auto const border = std::string{"* *"};
+    auto const fill = total_length - 2 * border.size() - text.size();
+    auto const fill_left = fill / 2;
+    auto const fill_right = fill - fill_left;
+    return border + std::string(fill_left, ' ') + text +
+           std::string(fill_right, ' ') + border;
+  };
+  std::cout << '\n'
+            << banner_string("Goto-Harness", CBMC_VERSION) << '\n'
+            << align_center_with_border("Copyright (C) 2019") << '\n'
+            << align_center_with_border("Diffblue Ltd.") << '\n'
+            << align_center_with_border("info@diffblue.com")
```

Comment:
We should likely have this `align_center_with_border` code for everyone...

Reply:
https://github.com/diffblue/cbmc/pull/3920